src/classes/PREQ_PreventPPlanParent_TDTM.cls

```diff
@@ -0,0 +1,71 @@
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Plan Requirement
+* @group-content ../../ApexDocContent/PlanRequirements.htm
+* @description Ensure that child plan requirement does not have program plan associated
+*/
+public with sharing class PREQ_PreventPPlanParent_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description If unique Program Plan for plan requirements validation enabled.
+    */
+    private static boolean stgUniqueProgramPlanEnabled = UTIL_CustomSettingsFacade.getSettings().Validate_Program_Plan_for_Nested_PR__c;
+
+    /*******************************************************************************************************
+    * @description Display an error when populating program plan on a child plan requirement
+    * @param listNew the list of Plan Requirement from trigger new.
+    * @param listOld the list of Plan Requirement from trigger old.
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @param objResult the describe for Plan Requirement
+    * @return dmlWrapper.
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
+            TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert
+            || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+            for (Plan_Requirement__c planReq : [SELECT ID,
+                                                    Program_Plan__c,
+                                                    Plan_Requirement__c
+                                                FROM Plan_Requirement__c
+                                                WHERE ID in :newlist]) {
+                if (stgUniqueProgramPlanEnabled == true
+                        && planReq.Plan_Requirement__c != null
+                        && planReq.Program_Plan__c != null) {
+                    //throw an error
+                    throw new ERR_ExceptionHandler.NestedPlanRequirementPPException(Label.nestedPlanRequirementPPExp);
```

Comment:
@xinnanlisf @lifewithryan I think we're getting exception emails here because we're throwing an exception. I think if we add an error like planReq.addError(Label.whateverTheLabelIs), it should be handled differently.

cc @ceiroa

Reply:
@robertbwatson 
I don't think this would work since planReq is not available in trigger context.

Reply:
@lifewithryan too

Reply:
@xinnanlisf Actually, I think I missed this but why are we querying the Plan Requirement? We should be able to just loop over the newlist since we're only accessing fields on the Plan Requirement.

Reply:
I think we had to query the object before when we were checking for proper nesting, but it seems like we can get away with not needing to query here.

Reply:
@robertbwatson that's a fair point.

Reply:
For future reference, we can even map it to the old list. e.g. Contact contactInContext = (Contact)oldmap.get(c.ID)

Reply:
@xinnanlisf  @lifewithryan Depending what we decide here may also impact my PR on Time Fields, as I chose to use the addError() method: https://github.com/SalesforceFoundation/EDA/pull/837/files#diff-c9996d678f52cbc57cee482c01f4d94cR140